File: flybanner/src/main/java/com/xia/flybanner/listener/FBPageChangeListener.java
package com.xia.flybanner.listener;

import android.widget.ImageView;

import java.util.ArrayList;

import androidx.recyclerview.widget.RecyclerView;

/**
 * @author weixia
 * @date 2019/4/16.
 */
public class FBPageChangeListener implements OnPageChangeListener {
    private ArrayList<ImageView> mPointViews;
    private int[] mPageIndicatorId;
    private OnPageChangeListener mOnPageChangeListener;

    public void setPageIndicator(final ArrayList<ImageView> pointViews, final int[] pageIndicatorId) {
        this.mPointViews = pointViews;
        this.mPageIndicatorId = pageIndicatorId;
    }

    @Override
    public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
        if (mOnPageChangeListener != null) {
            mOnPageChangeListener.onScrollStateChanged(recyclerView, newState);
        }
    }

    @Override
    public void onScrolled(RecyclerView recyclerView, int dx, int dy) {
        if (mOnPageChangeListener != null) {
            mOnPageChangeListener.onScrolled(recyclerView, dx, dy);
        }
    }

    @Override
    public void onPageSelected(int index, boolean isLastPage) {
        if (mPointViews != null && mPageIndicatorId != null) {
            final int size = mPointViews.size();
            for (int i = 0; i < size; i++) {
                mPointViews.get(index).setImageResource(mPageIndicatorId[1]);
                if (index != i) {
                    mPointViews.get(i).setImageResource(mPageIndicatorId[0]);
                }
            }
        }
        if (mOnPageChangeListener != null) {
            mOnPageChangeListener.onPageSelected(index, isLastPage);
        }
    }

    public void setOnPageChangeListener(OnPageChangeListener onPageChangeListener) {
        this.mOnPageChangeListener = onPageChangeListener;
    }
}

File: flybanner/src/main/java/com/xia/flybanner/listener/FBPageChangeListener.kt
package com.xia.flybanner.listener

import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import java.util.*

/**
 * @author weixia
 * @date 2019/4/16.
 */
class FBPageChangeListener : OnPageChangeListener {
    private var pointViews: ArrayList<ImageView>? = null
    private var pageIndicatorId: IntArray? = null
    private var pageChangeListener: OnPageChangeListener? = null

    fun setPageIndicator(pointViews: ArrayList<ImageView>, pageIndicatorId: IntArray) {
        this.pointViews = pointViews
        this.pageIndicatorId = pageIndicatorId
    }

    override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
        pageChangeListener?.onScrollStateChanged(recyclerView, newState)
    }

    override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
        pageChangeListener?.onScrolled(recyclerView, dx, dy)
    }

    override fun onPageSelected(index: Int, isLastPage: Boolean) {
        pointViews?.apply {
            val size = this.size
            for (i in 0 until size) {
                this[index].setImageResource(pageIndicatorId?.get(1) ?: 0)
                if (index != i) {
                    this[i].setImageResource(pageIndicatorId?.get(0) ?: 0)
                }
            }
        }
        pageChangeListener?.onPageSelected(index, isLastPage)
    }

    fun setOnPageChangeListener(onPageChangeListener: OnPageChangeListener?) {
        this.pageChangeListener = onPageChangeListener
    }
}
